src/caja-window.c

```diff
@@ -1364,18 +1364,20 @@ load_view_as_menu (CajaWindow *window)
     G_GNUC_END_IGNORE_DEPRECATIONS;
     window->details->view_as_radio_action = NULL;
 
-    /* Add a menu item for each view in the preferred list for this location. */
-    /* Start on 1, because extra_viewer gets index 0 */
-    for (node = window->details->short_list_viewers, index = 1;
-            node != NULL;
-            node = node->next, ++index)
-    {
-        /* Menu item in View menu. */
-        add_view_as_menu_item (window,
-                               CAJA_MENU_PATH_SHORT_LIST_PLACEHOLDER,
-                               node->data,
-                               index,
-                               merge_id);
+    if (g_list_length (window->details->short_list_viewers) > 1) {
+        /* Add a menu item for each view in the preferred list for this location. */
+        /* Start on 1, because extra_viewer gets index 0 */
```

Comment:
I think you can cut / paste the line 1333 `int index;` to here to avoid the new warning:

`[src/caja-window.c:1333]: (style) The scope of the variable 'index' can be reduced.`

Reply:
fixed.